pkg/products/monitoring/reconciler.go

```diff
@@ -261,8 +262,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, installation *integreatlyv1a
 		return phase, err
 	}
 	*/
-
-	phase, err = r.reconcileAlertManagerConfigSecret(ctx, serverClient)
+	phase, err = monitoringcommon.ReconcileAlertManagerSecrets(ctx, serverClient, r.installation, r.Config.GetOperatorNamespace(), alertManagerRouteName)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
```

Comment:
The above function can also return `Awaiting components` Will the HandleError here put RHMI into error state in that case?

Reply:
Good catch , copy and past error should be logic should be https://github.com/integr8ly/integreatly-operator/blob/master/pkg/products/monitoring/reconciler.go#L265-L273